ring/token_generator.go

```diff
@@ -7,8 +7,9 @@ import (
 )
 
 type TokenGenerator interface {
-	// GenerateTokens generates unique tokensCount tokens, none of which clash
-	// with the given takenTokens. Generated tokens are sorted.
+	// GenerateTokens generates tokensCount unique tokens, none of which clash with the given takenTokens.
+	// Generated tokens are sorted.
+	// If it is not possible to generate the requested tokensCount tokens, GenerateTokens panics.
```

Comment:
There's no need to specify this. All implementations should be able to generate the requested count of tokens. 

The implementation you're introducing also does that, we have a test that ensures that it doesn't panic.